sdk/core/Azure.Core/src/ETag.cs

```diff
@@ -72,28 +91,60 @@ public override int GetHashCode()
         ///
         /// </summary>
         /// <returns>The string representation of this <see cref="ETag"/>.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString()
         {
             return _value ?? "<null>";
         }
 
+        /// <summary>
+        /// Returns the string representation of the <see cref="ETag"/>.
+        /// </summary>
+        /// <param name="format">A format string. Valid values are "G" for standard format and "H" for header format.</param>
+        /// <returns>The formatted string representation of this <see cref="ETag"/>. This includes outer quotes and the W/ prefix in the case of weak ETags.</returns>
+        /// <example>
+        /// <code>
+        /// ETag tag = ETag.Parse("\"sometag\"");
+        /// Console.WriteLine(tag.ToString("G"));
+        /// // Displays: sometag
+        /// Console.WriteLine(tag.ToString("H"));
+        /// // Displays: "sometag"
+        /// </code>
+        /// </example>
+        public string ToString(string format)
+        {
+            if (_value == null)
+            {
+                return "<null>";
+            }
+            return format switch
+            {
+                HeaderFormat => _preserveRawValue ? _value : $"{QuoteString}{_value}{QuoteString}",
+                DefaultFormat => ToString(),
```

Comment:
this might be no important, but by calling ToString you are doing the _value null check twice.

Reply:
Yes, I though this was the better trade off over re-implementing the existing `ToString()` logic here.

Reply:
I am OK with that, but to be clear, the "re implementation" would be => _value; (as you already checked for null above).

Reply:
Thinking more about it, we should probably just refactor `ToString()` as a lambda to `ToString("G")` and return _value here.